From 9af50392f117b8d3926a25e1b9ee1e7562a38285 Mon Sep 17 00:00:00 2001
From: Ken Ma <make@marvell.com>
Date: Mon, 25 Jun 2018 10:17:27 +0800
Subject: [PATCH 0669/1051] arm64: dts: a7k/a8k: add spi direct access mapping
 for DB boards

In ATF, spi direct mapping window of [0xf9000000, 0xf9ffffff] is added
for a70x0/a70x0_cust(CP0 SPI1 CS0) and a80x0/a80x0_mcbin(CP1 SPI1 CS0).

This patch adds the spi direct access mapping for a70x0/a80x0 DB and
a80x0 mcbin boards to enable direct write and get write performance
improvement about 80%.

Change-Id: I6cceac0e5fb53869c65f04ba64b3ccc577365713
Signed-off-by: Ken Ma <make@marvell.com>
Signed-off-by: Igal Liberman <igall@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-7040-db.dtsi   | 9 +++++++++
 arch/arm64/boot/dts/marvell/armada-8040-db.dtsi   | 9 +++++++++
 arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts | 9 +++++++++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi     | 2 +-
 4 files changed, 28 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
index f9a5728430a9..30aadbe53343 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
@@ -69,6 +69,13 @@
 	non-removable;
 };
 
+&cp0_config_space {
+	ranges = /* internal regs */
+		 <0x0 0x0 0xf2000000 0x2000000>,
+		 /* SPI1-DEV0 */
+		 <0x2000000 0 0xf9000000 0x1000000>;
+};
+
 &cp0_i2c0 {
 	expander0: pca9555@21 {
 		compatible = "nxp,pca9555";
@@ -101,6 +108,8 @@
 };
 
 &cp0_spi1 {
+	reg = <0x700680 0x50>,          /* control */
+	      <0x2000000 0x1000000>;    /* CS0 */
 	spi-flash@0 {
 		#address-cells = <0x1>;
 		#size-cells = <0x1>;
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
index 6eaa96d64e0e..f9fb1ab96511 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
@@ -158,7 +158,16 @@
 	phy-names = "usb";
 };
 
+&cp1_config_space {
+	ranges = /* internal regs */
+		 <0x0 0x0 0xf4000000 0x2000000>,
+		 /* SPI1-DEV0 */
+		 <0x2000000 0 0xf9000000 0x1000000>;
+};
+
 &cp1_spi1 {
+	reg = <0x700680 0x50>,          /* control */
+	      <0x2000000 0x1000000>;    /* CS0 */
 	status = "disabled";
 
 	spi-flash@0 {
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
index 9319d0be507a..6710ef89f704 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-mcbin.dts
@@ -294,6 +294,13 @@
 	status = "okay";
 };
 
+&cp1_config_space {
+	ranges = /* internal regs */
+		 <0x0 0x0 0xf4000000 0x2000000>,
+		 /* SPI1-DEV0 */
+		 <0x2000000 0 0xf9000000 0x1000000>;
+};
+
 &cp1_ethernet {
 	status = "okay";
 };
@@ -378,6 +385,8 @@
 &cp1_spi1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&cp1_spi1_pins>;
+	reg = <0x700680 0x50>,          /* control */
+	      <0x2000000 0x1000000>;    /* CS0 */
 	status = "okay";
 
 	spi-flash@0 {
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 4678caefc5ee..ded455ee43f7 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -33,7 +33,7 @@
 	interrupt-parent = <&CP110_LABEL(icu)>;
 	ranges;
 
-	config-space {
+	CP110_LABEL(config_space): config-space {
 		#address-cells = <1>;
 		#size-cells = <1>;
 		compatible = "simple-bus";
-- 
2.17.1

